python/lsst/ts/ledprojector/ledprojector_csc.py

```diff
@@ -0,0 +1,336 @@
+                # Turn off the led, if connected
+                if self.led_connected:
+                    self.log.warning(
+                        "Going to fault while connected to the led controller; "
+                        "trying to turn the led off immediately."
+                    )
+                    try:
+                        await self.led_model.turn_led_off(
+                            force=True, wait=False, reason="CSC is going to FAULT state"
+                        )
+                    except Exception as e:
+                        self.log.error(
+                            f"Failed to turn led off; please turn it off manually: {e!r}"
+                        )
+                else:
+                    self.log.warning(
+                        "Going to fault and not connect to the led controller; "
+                        "please turn the led off manually."
+                    )
+        else:
+            self.should_be_connected = False
+            if self.led_connected and self.led_model.led_was_on:
+                try:
+                    await self.led_model.turn_led_off(
+                        force=True,
+                        wait=False,
+                        reason=f"CSC is going to state {self.summary_state!r}",
+                    )
+                except Exception as e:
+                    self.log.warning(
+                        f"Going to state {self.summary_state!r} but failed to turn off led: {e!r}; "
+                        "please turn it off manually."
+                    )
+            await self.disconnect_led()
+
+
+    async def connect_led(self):
+        """Connect to the led controller LabJack and get status.
+
+        Raises
+        ------
+        asyncio.TimeoutError
+            If it takes longer than self.config.led.connect_timeout
+        """
+        await self.disconnect_led()
+        if self.led_model is None:
+            self.led_model = LEDModel(
+                config=self.config.led,
+                csc=self,
+                log=self.log,
+                status_callback=self.status_callback,
+                simulate=self.simulation_mode != 0,
+                make_connect_time_out=self.led_make_connect_time_out,
+            )
+        await asyncio.wait_for(
+            self.led_model.connect(), timeout=self.config.led.connect_timeout
+        )
+
+    async def disconnect_led(self):
+        try:
+            # Don't use self.led_connected because that can be false
+            # even if a basic connection exists (before status seen)
+            if self.led_model is None:
+                return
+            await self.led_model.disconnect()
+            # Delete the led model because the config may change.
+            self.led_model = None
+        except Exception as e:
+            self.log.warning(f"Failed to disconnect led; continuing: {e!r}")
+
+    async def start(self):
```

Comment:
I would move the evt lines into the model when you connect/disconnect with setting connected to either true/false respectively and then you can remove this method here.